### PR TITLE
Add Apache Flink

### LINKED ISSUE #850: Add flink
Hello,

I and other members of the flink community have been working on refining the docker image with the goal of making it an official image, and we expect to soon start a PR to get it out.
https://github.com/apache/flink/blob/master/flink-contrib/docker-flink/Dockerfile

Notice that we are still cleaning stuff, but I wanted to start the discussion here to get some early feedback.

### PR DESCRIPTION
**Pull request by patricklucas**
Associated official-images PR: https://github.com/docker-library/official-images/issues/2895

Closes #850

### CODE REVIEW

File: flink/get-help.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Community & Project Info](https://flink.apache.org/community.html)
```
Comment (tianon) [inline flink/get-help.md]: There's a `[` missing at the beginning here.

File: flink/content.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+```console
+$ docker run --name flink_local -p 8081:8081 -t flink local
+```
+
+Then with a web browser go to `http://localhost:8081/` to see the Flink Web Dashboard (adjust the hostname for your Docker host).
+
+To use Flink, you can submit a job to the cluster using the Web UI or you can also do it from a different Flink container, for example:
+
+```console
+$ docker run -t flink flink run -m <jobmanager:port> -c <your_class> <your_jar> <your_params>
```
Comment (tianon) [inline flink/content.md]: Might be a good idea to add `--rm` to this one, since it's not long-running. :+1:

**Comment by tianon** [comment]
Apparently copy-pasting links is hard -- https://github.com/docker-library/official-images/pull/2895#issuecomment-300229587 is what I meant to link to. :sweat:

File: flink/get-help.md (hunk 1 above)
Comment (patricklucas) [inline flink/get-help.md]: good catch

**Comment by patricklucas** [comment]
Changes made. Thanks for the review. :)

**Pull request merged**